pyblish_lite/control.py

```diff
@@ -58,40 +66,107 @@ def __init__(self, parent=None):
             "ordersWithError": set()
         }
 
-    def reset(self):
-        """Discover plug-ins and run collection"""
+    def prepare_for_reset(self):
+        self.validated = False
+        self.extracted = False
+        self.publishing = False
+
         self.context = pyblish.api.Context()
-        self.plugins = pyblish.api.discover()
+        self.context.data["optional"] = False
+        self.context.data["publish"] = True
 
-        self.was_discovered.emit()
+        self.context.data["label"] = 'Context'
```

Comment:
Does this support customising the context using a collector? It looks hardcoded.

```py
from pyblish import api

class MyCollector(api.ContextPlugin):
   order = api.CollectorOrder

   def process(self, context):
      context.data["label"] = "MyCustomLabel"
```

This should work.

Reply:
It is hardcoded (same as QML) and it is initial label - Your collector would work and override that...